### ISSUE TITLE
[GTK+3] Keyboard layout indicator text in the panel is always white

### ISSUE THREAD

**Issue by GiedriusS** [mate-desktop/libmatekbd#28]
As seen in this screenshot https://i.imgur.com/VRBC772.jpg the indicator text that shows the current layout is white. And it is always white regardless of the theme being used. This happens _only_ on gtk3 builds. @raveit65 also confirmed this. libmatekbd version:
[*] libmatekbd-1.12.0_1             MATE keyboard library

**Comment by monsta** [comment]
Workaround:
```
gsettings set org.mate.peripherals-keyboard-xkb.indicator foreground-color '0 0 0'
```
This will set the font to be always black. If you don't plan changing theme to dark this should be ok :smile:

**Comment by raveit65** [comment]
nice finder, in a ideal world it should read theme fg color, but a gsettings key is sufficient :)

**Comment by raveit65** [comment]
possible fix
https://git.gnome.org/browse/libgnomekbd/commit/?id=ab411c5084e1816df56f97351132329aeb4123be

**Comment by monsta** [comment]
Yes, it might be. I've already seen it before but didn't like the "hack" word... but we can try it since there's no other fix around...

**Comment by monsta** [comment]
Nice, it's indeed fixed now, thanks @XRevan86 :smile:
The font has a bit weird rendering (somewhat blurred), but the color is correct for both light and dark themes.

**Comment by monsta** [comment]
So I saw this fixed in Debian Testing with GTK+ 3.20, but now I see it in Ubuntu MATE 16.10 with GTK+ 3.18... tried Menta, TraditionalGreen and other themes, the font is still white. :confused:

**Comment by monsta** [comment]
@flexiondotorg: can you reproduce this in Ubuntu 16.10?

**Comment by monsta** [comment]
@flexiondotorg: I see it now in Ubuntu 16.10 - only with Radiant-MATE though. Seems to be fixed in other themes, at least for 3.20.

**Comment by oz123** [comment]
This bug still exists for me in mate-1.16 built on gentoo.

**Comment by oz123** [comment]
As a work around for this annoying issue, I can suggest the following remedy:

edit `foreground-color` in `org->mate->desktop->peripherals->keyboard->indicator`.

Change it to `0 0 0`.

**Comment by monsta** [comment]
Which theme do you use?

**Comment by oz123** [comment]
This happened with GTK+3.18 and GTK+3.20, with the following themes:

 Menta, TraditionalGreen, TraditionalOK, GreenSubmarine, BlueMenta and Adwaita.

**Comment by raveit65** [comment]
> This happened with GTK+3.18 and GTK+3.20, with the following themes:

> Menta, TraditionalGreen, TraditionalOK, GreenSubmarine, BlueMenta and Adwaita.

Not here in fedora with gtk+-3.20/22, Changing to a theme with another bg color for the panel change the font color like it should.

**Comment by monsta** [comment]
Using latest mate-themes is the key. Here I have GTK+ 3.18.9, and I use the latest 3.18 themes, and I don't see this issue.

**Comment by vkotovv** [comment]
Btw, issue is still present in Ubuntu 16.04 LTS, because it uses GTK+ 3.18. Had to use this workaround with `foreground-color'

**Comment by monsta** [comment]
If you use the default MATE packages in 16.04, then it's GTK+2 build. I'm not sure we'll be able to fix this for GTK+2.

**Comment by vkotovv** [comment]
No, I'm using [mate 1.18 ppa](https://launchpad.net/~jonathonf/+archive/ubuntu/mate-1.18)

**Comment by monsta** [comment]
Ok, then a theme fix is needed.

**Comment by raveit65** [comment]
> No, I'm using mate 1.18 ppa

And which theme are you using?
I fixed that for Mate-Radiant theme https://github.com/flexiondotorg/ubuntu-mate-themes/commit/61217fea9b1ac963eadf953226254deb25376a16
If you use another theme you need to add something like this
```
/* needed for keyboard layout indicator applet */
.mate-panel-menu-bar {
    color: @theme_fg_color;
}
```

Note, .**mate-panel-menu-bar** is the main style class for the panel.

**Comment by vkotovv** [comment]
I use Radiance (as it stated through Appearance Preferences), I guess Mate-Radiant is the other name for that. How can I make use of your fix?

**Comment by raveit65** [comment]
No, Mate-Radiant is based on Radiance but adjusted for Mate.
Create ~.config/gtk-3.0/gtk.css with that content.
```
/* needed for keyboard layout indicator applet */
.mate-panel-menu-bar {
    color: @fg_color;
}
``` 
and reload session for restarting mate-settings-daemon.
As you can see
/usr/share/themes/Radiance/gtk-3.20/gtk-main.css  use another color definition as Radiant-Mate.

**Comment by monsta** [comment]
@flexiondotorg: I can confirm Radiance still has this issue, not only in 16.04, but also in 16.10 and 17.04.

**Comment by monsta** [comment]
Happens with GTK+ 3.14 as well, but with a weird effect: the indicator has light green + white colors (with a gradient), at least in my VM. This looks quite funny with BlackMATE theme. :slightly_smiling_face: 

Reopening, we need to get this fixed for any theme and any GTK+ version...

**Comment by N0rbert** [comment]
From my [wrongly reported bug](https://github.com/mate-desktop/mate-indicator-applet/issues/23):

#### Steps to reproduce:
1. Install Ubuntu 18.04 LTS
2. Install `ubuntu-mate-desktop`
3. Set light theme from `mate-control-center` Appearance - for example darkroom:
![darkroom](https://user-images.githubusercontent.com/309456/34502316-ffb40864-f023-11e7-8ee5-fb1ecf34c154.png)

#### Expected results:
* keyboard layout indicator is visible on light and dark themes

#### Actual results:
* keyboard layout indicator has low contrast on light themes

#### MATE general version
1.18.4

#### Linux Distribution
Ubuntu 18.04 LTS

#### Link to downstream report of your Distribution
Issue filed to the launchpad as [bug 1728711](https://bugs.launchpad.net/bugs/1728711).

**Comment by raveit65** [comment]
> Reopening, we need to get this fixed for any theme and any GTK+ version...

Any one working on that?
This is fixed in Mate themes an Ubuntu Mate themes.
I see no reason to let a report open if no one is working on that.

**Comment by tamplan** [comment]
I can't reproduce this bug on fedora 30 with libmatekbd 1.23 (master) with light theme:
![Capture du 2019-06-16 22-19-19](https://user-images.githubusercontent.com/50292710/59569101-3a9e1700-9085-11e9-88e5-a6596758fa5d.png)

Regards

**Comment by oratosquilla-oratoria** [comment]
Still can reproduce with any non-standard light theme (which do not have the fix from https://github.com/mate-desktop/libmatekbd/issues/28#issuecomment-297430508).
MATE v1.24.1.
